platform/pensando/sonic-platform-modules-dpu/dpu/utils/dpu_pensando_util.py

```diff
@@ -0,0 +1,311 @@
+        return f'python{python_version}'
+    except:
+        return 'python3.9'
+
+def fetch_dpu_files():
+    docker_id = apiHelper.get_dpu_docker_imageID()
+    cmd = "sudo docker cp {}:/tmp/fru.json /home/admin".format(docker_id)
+    run_cmd(cmd)
+    cmd = "sudo docker cp {}:/nic/bin/cpldapp /home/admin".format(docker_id)
+    run_cmd(cmd)
+    cmd = "sudo docker cp {}:/nic/lib/libpal.so /home/admin".format(docker_id)
+    run_cmd(cmd)
+    cmd = "sudo docker cp {}:/nic/lib/liblogger.so /home/admin".format(docker_id)
+    run_cmd(cmd)
+    cmd = "sudo docker cp {}:/nic/etc/VERSION.json /home/admin".format(docker_id)
+    run_cmd(cmd)
+    cmd = "sudo docker cp {}:/usr/bin/mmc /usr/local/bin".format(docker_id)
+    run_cmd(cmd)
+
+def set_onie_version():
+    version = ''
+    try:
+        cmd = 'cat /host/machine.conf | grep -i onie_version'
+        output = run_cmd(cmd)
+        version = output.split('=')[1]
+    except:
+        pass
+    if version == '':
+        try:
+            cmd = "fw_printenv onie_version"
+            output = run_cmd(cmd)
+            version = output.split('=')[1]
+        except:
+            version = "Not Available"
+            pass
+    try:
+        fru_file = "/home/admin/fru.json"
+        data = json.load(open(fru_file))
+        data["onie_version"] = version
+        with open(fru_file, "w") as json_file:
+            json.dump(data, json_file, indent=4)
+    except:
+        pass
+
+def cp_to_shared_mem():
+    api_helper_platform = apiHelper.get_platform()
+    platform = api_helper_platform if api_helper_platform != None else "arm64-elba-asic-r0"
+    cmd = "sudo cp /home/admin/fru.json /usr/share/sonic/device/{}/fru.json".format(platform)
+    run_cmd(cmd)
+    cmd = "sudo cp /home/admin/cpldapp /usr/share/sonic/device/{}/cpldapp".format(platform)
+    run_cmd(cmd)
+    cmd = "sudo cp /home/admin/libpal.so /usr/share/sonic/device/{}/libpal.so".format(platform)
+    run_cmd(cmd)
+    cmd = "sudo cp /home/admin/liblogger.so /usr/share/sonic/device/{}/liblogger.so".format(platform)
+    run_cmd(cmd)
+    cmd = "sudo cp /home/admin/VERSION.json /usr/share/sonic/device/{}/VERSION.json".format(platform)
+    run_cmd(cmd)
+
+def set_cpldapp():
+    cmd = "sudo cp /home/admin/cpldapp /usr/local/bin"
+    run_cmd(cmd)
+    cmd = "sudo cp /home/admin/libpal.so /lib/libpal.so"
+    run_cmd(cmd)
+    cmd = "sudo cp /home/admin/liblogger.so /lib/liblogger.so"
+    run_cmd(cmd)
+
+def cp_sonic_platform_helper():
+    python_dir = get_python_lib_dir()
+    lib_path = f"/usr/local/lib/{python_dir}/dist-packages/"
+    sonic_platform_path = lib_path + "sonic_platform/"
+    sonic_platform_helper_path = sonic_platform_path  + "helper.py"
+    if not os.path.exists(sonic_platform_helper_path):
+        sonic_platform_old_path = "/usr/lib/python3/dist-packages/sonic_platform"
+        cmd = "cp -r " + sonic_platform_old_path + " " + lib_path
+        run_cmd(cmd)
```

Comment:
Why is this here? Why is it needed? This (and using the Python version) is fragile code.

Reply:
Ack, addressed it.